Add support for RetroNet line things

RealDeuce opened the pull request:
0xdc (line count) and 0xdd (get line)

These are used by (at least) the slideshow.

RealDeuce committed:
Add support for RetroNet line things

0xdc (line count) and 0xdd (get line)

These are used by (at least) the slideshow.

thorpej commented:
Looks good at the quick glance I gave it this morning.  Thanks so much for doing this!  I'll get it merged sometime this week and a new release tagged!

RealDeuce commented:
I'm also working on the TCP client support.
The "TCP Server" API looks like gibberish though, so I likely won't be supporting that (whatever it does).

RealDeuce committed:
Implement TCP client interface

This implements the following RetroNet commands:
0xD0: rn_TCPOpen
0xD1: rn_TCPHandleClose
0xD2: rn_TCPHandleSize
0xD3: rn_TCPHandleRead
0xD4: rn_TCPHandleWrite

The socket is non-blocking with Nagle disabled, and errors that do
not indicate the socket is unusable are suppressed.

This works with the CP/M telnet clients.

thorpej reviewed (approved):
Looks good.  I'll make a couple of small stylistic tweaks after merging.  Thanks!
